src/compat/posix/fnmatch/fnmatch.c

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief fnmatch_function, now working with two flags - FNM_MATCH and FNM_PATHNAME
+ * @date 20.12.18
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <fnmatch.h>
+
+static int bracket_check(const char *pattern) {
+	while (*pattern != ']') {
+		if (*pattern == '/') {
+			return 0;
+		}
+		pattern++;
+	}
+	return 1;
+}
+
+static int fnmatch_string(const char *pattern, const char *string, int flags, int asterisk_pat) {
+	int flag = 0;
+	int denial_flag = 0;
+	while (*pattern != '\0') {
+
+		if((*string == '/') && asterisk_pat && (flags == FNM_PATHNAME)) {
+			return 0;
+		}
+
+		if ((*string == '\0') && (*pattern != '*')) {
+			return FNM_NOMATCH;
+		}
+
+		switch (*pattern) {
+
+			case '?':
+				if ((flags == FNM_PATHNAME) && (*string == '/')) {
+					return FNM_NOMATCH;
+				}
+				break;  
+
+			case '*':
+				pattern++;
+				if (*pattern == '\0') {
```

Comment:
For `FNM_PATHNAME` zero should be returned only if there are no slashes in the rest of `string`